storage/database.go

```diff
@@ -100,7 +99,7 @@ type db struct {
 
 	// Contains an entry to all shards for fast shard lookup, an
 	// entry will be nil when this shard does not belong to current database
-	shards []databaseShard
+	shards map[uint32]databaseShard
```

Comment:
Specifically did not want to use a map here as map access is significantly slower than array lookup.  Can you revert to `[]databaseShard`?